unmock/core/options.py

```diff
@@ -41,6 +46,45 @@ def __init__(self, save: Union[bool, List[str]] = False, unmock_host: str = UNMO
             persistence = FSPersistence(self.token)
         self.persistence = persistence
 
+    def get_token(self) -> Optional[str]:
+        """
+        Fetches a new access token from the unmock server or predisposed access token if it is still valid.
+        Throws RuntimeError on logical failures with unexpected responses from the Unmock host.
+        """
+        url = "{scheme}://{host}:{port}".format(scheme=self.scheme, host=self.unmock_host, port=self.unmock_port)
+        access_token = self.persistence.load_auth()
+        if access_token is not None:  # If we already have an access token, let's see we can still ping with it
+            if self._validate_access_token(access_token):  # We can ping, all's good in the world!
+                return access_token
+
+        # Otherwise, we need to get a new token
+        refresh_token = self.persistence.load_token()
+        if refresh_token is None:
+            return  # Continue with the public API ('/y/' version)
+        response = requests.post("{url}/token/access".format(url=url), json={"refreshToken": refresh_token})
+        if response.status_code == HTTPStatus.OK:
+            new_access_token = response.json().get("accessToken")
+            if new_access_token is None:  # Response was not present..?
+                raise RuntimeError("Incorrect server response: did not get accessToken")
```

Comment:
At some point library users may want to catch specific `UnmockException`s in their code but using `RuntimeError` is probably ok for now.

Reply:
True. It's an odd case because if any of those `RuntimeError` are thrown, then there's probably something wrong in the cloud :thinking: